src/responder/ssh/ssh_cmd.c

```diff
@@ -98,11 +99,21 @@ static errno_t ssh_cmd_get_user_pubkeys(struct cli_ctx *cli_ctx)
         goto done;
     }
 
-    subreq = cache_req_user_by_name_attrs_send(cmd_ctx, cli_ctx->ev,
-                                               cli_ctx->rctx,
-                                               cli_ctx->rctx->ncache, 0,
-                                               cmd_ctx->domain,
-                                               cmd_ctx->name, attrs);
+    data = cache_req_data_name_attrs(cmd_ctx, CACHE_REQ_USER_BY_NAME,
+                                     cmd_ctx->name, attrs);
+    if (data == NULL) {
+        ret = ENOMEM;
+        goto done;
+    }
+
+    cache_req_data_set_bypass_cache(data, true);
```

Comment:
While I still think it is a good idea to make sure the the ssh keys are all still valid, always bypassing the cache might be a considerable performance drop for various use-cases. Because I realized that during an authentication attempt with ssh-keys we would unconditionally refresh the user object here and then in the PAM responder during the access control step.

Until we have a better coordination between the responders I think it might be better to postpone this patch and document that ssh-key will not be deleted immediately in the cache. As an alternative a new option can be added to switch the by-passing in the ssh responder on and off. But this option would become somewhat obsolete when we have a good coordination of the unconditional refreshes in the responders.